### ISSUE TITLE
The screen will be black for one second on IOS platform

### ISSUE THREAD

**Issue by jjboy91** [jonbhanson/flutter_native_splash#463]
Hello,

I'm experiencing the same issue signaled in #219
After the app's launch, the FlutterNativeSplash appears, then a black screen, then the first screen of the app.

```
void main() async {
  final binding = WidgetsFlutterBinding.ensureInitialized();

  FlutterNativeSplash.preserve(widgetsBinding: binding);

  binding.deferFirstFrame();
  binding.addPostFrameCallback((_) async {
    BuildContext context = binding.renderViewElement;
    if (context != null) {
      await _precacheAssets(context);
    }

    // FlutterNativeSplash.remove(); // Not working, tried moving it to before the first screen's call but same behaviour
    binding.allowFirstFrame();
  });
  
  runApp(...);
}

Future<bool> _precacheAssets(BuildContext context) async {
  await precacheImage(
      Image.asset('assets/images/background.png').image, context);

  return Future<bool>.value(true);
}

```

**Configuration**

 flutter_native_splash config :
```
flutter_native_splash:
   color: "#8bc34a"
   background_image: "assets/images/background.png"
```

**Device **
 - Device: iPad Mini 2
 - OS: 12.5.6

**Additional context**
I tried without the precacheAssets part and it's the same problem.

What did I missed ? 

Thanks

**Comment by jonbhanson** [comment]
You don't need to use both `FlutterNativeSplash.preserve` and `binding.deferFirstFrame`, because under the hood `FlutterNativeSplash.preserve` is calling `binding.deferFirstFrame`.  Also, `FlutterNativeSplash.remove` makes a call to `binding.allowFirstFrame`.  If you remove these duplicate calls do you still have the same problem?

**Comment by jjboy91** [comment]
Alright my bad, so I did the following : 

```
final binding = WidgetsFlutterBinding.ensureInitialized();
FlutterNativeSplash.preserve(widgetsBinding: binding);
binding.addPostFrameCallback((_) async {
  BuildContext context = binding.renderViewElement;
  if (context != null) {
    await _precacheAssets(context);
  }

  FlutterNativeSplash.remove();
});
```
Except than I am experiencing the same behaviour, black scren for 1-2 seconds then the first screen.

**Comment by jonbhanson** [comment]
@jjboy91 can you send me a stripped-out version of your app so that I can diagnose this problem?

**Comment by federico-amura-kenility** [comment]
I am having the same black screen on ios

**Comment by sirawatGG** [comment]
@jonbhanson
dart - sdk: ">=2.18.0 <3.0.0"
flutterSdkVersion": "3.3.7
lib-version flutter_native_splash: ^2.2.16

**Comment by mike-robinson** [comment]
I had the same problem when using a background_image. On startup, there was always 1-2 seconds of black screen. It turned out that the image was too large. It was 2234x3456 and was basically a gradient from top to bottom. I changed it to 10 x 3456 and the black screen no longer shows up. There are some references to large launch images on apple developer and they  must apply to launch background images as well. https://developer.apple.com/forums/thread/666036

**Comment by mike-robinson** [comment]
For complex apps, my previous experiment does not do the trick by itself.  (I'm not even sure if the image size was relevant) But my main() has a lot of initialization that needs to be done before the app launches. Originally, main looked something like this

`      
WidgetsBinding widgetsBinding = WidgetsFlutterBinding.ensureInitialized();
FlutterNativeSplash.preserve(widgetsBinding: widgetsBinding);

... other initialization calls and then runApp
`

Once I moved the preserve call after all the other initialization calls, the black screen went away.

`      
WidgetsBinding widgetsBinding = WidgetsFlutterBinding.ensureInitialized();
.... other initialization calls here

FlutterNativeSplash.preserve(widgetsBinding: widgetsBinding);

and then runApp
`
This was tested by pushing a test version to the app store.

**Comment by jonbhanson** [comment]
Thanks @mike-robinson for the diagnostic info.  If anyone still has a problem after trying Mike's solution, feel free to open a new ticket.

**Comment by sirawatGG** [comment]
@mike-robinson Sorry, I tried your solution, it doesn't work, open new ticket [here](https://github.com/jonbhanson/flutter_native_splash/issues/477)
